gradle-spaghetti-plugin/src/main/java/com/prezi/spaghetti/gradle/GenerateHeaders.java

```diff
@@ -1,21 +1,36 @@
 package com.prezi.spaghetti.gradle;
 
+import com.google.common.collect.Sets;
+import com.prezi.spaghetti.bundle.internal.DependentFiles;
 import com.prezi.spaghetti.definition.ModuleConfiguration;
+import com.prezi.spaghetti.definition.ModuleConfigurationParser;
+import com.prezi.spaghetti.definition.ModuleDefinitionSource;
 import com.prezi.spaghetti.generator.Generators;
 import com.prezi.spaghetti.generator.HeaderGenerator;
 import com.prezi.spaghetti.generator.internal.DefaultGeneratorParameters;
 import com.prezi.spaghetti.generator.internal.InternalGeneratorUtils;
 import com.prezi.spaghetti.gradle.internal.AbstractDefinitionAwareSpaghettiTask;
+import com.prezi.spaghetti.gradle.internal.ModuleBundleLookup;
+import com.prezi.spaghetti.gradle.internal.SpaghettiExtension;
 import org.apache.commons.io.FileUtils;
+import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.OutputDirectory;
 import org.gradle.api.tasks.TaskAction;
 
 import java.io.File;
 import java.io.IOException;
+import java.util.Iterator;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 public class GenerateHeaders extends AbstractDefinitionAwareSpaghettiTask {
 	private File outputDirectory;
 
+	public GenerateHeaders() {
+		SpaghettiExtension extension = getProject().getExtensions().getByType(SpaghettiExtension.class);
+		setDependentModules(getProject().files(extension.getModuleDefinitionConfiguration()));
```

Comment:
This is not going to work. The task's constructor runs when the task is created. At that point the `build.gradle` is executing `apply plugin: "boxfish-..."`, so the `boxfish` extension is not configured. Thus `extension.getModuleDefinitionConfiguration()` will always return the default setting.